src/chapters/loops/studio.rst

```diff
@@ -40,51 +43,65 @@ Part A (Put dinner together)
 
-   a. Use string formatting to print something more interesting than "['chicken', 'rice', 'peas', 'juice', 'apple']" for the meal outputs.
-   b. Use an "array of arrays" to store the food options in a 'pantry'.
+1. Using a ``while`` loop, ask the user to select the number of meals to assemble. Validate the input to make sure it is an integer from 1 - 6.
+2. Save the result to the ``numMeals`` variable that we initialized to -1 at the beginning.
 
-.. image:: figures/array-of-arrays.png
-    :height: 300px
 
-Part B (Self-destruct system)
------------------------------
+Part C: Create a Password Generator
+-----------------------------------
 
-If the shuttle gets hijacked by space pirates, the astronauts can activate
-a self-destruct sequence to provide some drama for the viewers at home.
+Working and living aboard this amazing space shuttle requires you to pay the utmost attention to cybersecurity.
+Once you are done working on the meal system, you are prompted to create a new password that will be used for the next 24 hours.
+Having run out of strong password ideas, your shuttle captain has encouraged you to make a password generator for yourself.
 
-In order to prevent a rogue astronaut from activating the code, it takes
-*two* crew members to begin the countdown.  Each person must enter a
-different code, after which the computer will "zip" them together before
-overloading the engines.
+Write your code inside ``generatePassword()``.
 
-4. Construct a ``for`` loop that combines two strings together, alternating the characters from each source.
+1. Construct a ``for`` loop that combines the two strings together, alternating the characters from each source, and saves the combined string to the ``code`` variable.
 
    .. admonition:: Examples
 
-      #. If ``string = "1234"`` and ``otherString = "5678"``, then the output will be "15263748".
-      #. If ``code1 = "ABCDEF"`` and ``code2 = "notyet"``, then the output will be "AnBoCtDyEeFt".
-      #. If ``ka = "LoOt"`` and ``blam = "oku!"``, then the output will be ``"LookOut!"``.
+      #. If ``string1 = "1234"`` and ``string2 = "5678"``, then the output will be "15263748".
+      #. If ``string1 = "ABCDEF"`` and ``string2 = "notyet"``, then the output will be "AnBoCtDyEeFt".
+      #. If ``string1 = "LoOt"`` and ``string2 = "oku!"``, then the output will be "LookOut!".
+
+2. Return ``code``.
+
+Part D: Put it all together!
+----------------------------
+
+Finally, turn your attention to ``runProgram()``.
 
-   `Code it at replt.it <https://repl.it/@launchcode/LoopstudiopartB>`__
+1. Write a ``for`` loop that uses ``mealPrintout()`` to format the printing of each meal in ``mealsFor6``.
+2. Write another ``for`` loop that uses ``mealPrintout()`` to format the printing of each meal in ``mealsForX``.
+3. Initialize the two strings, ``password1`` and ``password2``.
+4. Run your program to see how it works!
 
-Part C (Refinements)
---------------------
+Checking Your Work
+------------------
```

Comment:
Maybe add a link to Courtney's appendix here

Reply:
i think it wouldn't hurt.

Reply:
Got it!